### ISSUE TITLE
Implement a plugin audit command

### ISSUE THREAD

**Issue by lbb** [GoogleContainerTools/krew#31]
## Background:
We are currently not linting or verifying submissions for the krew index. We should provide a lint tool that is equal to the homebrew `audit` command.

## User Story:

```text
$ krew audit --source=./foo.yaml
Check Name Convention: OK
Validate Manifest: OK
Check Install: OK
Test Install and Uninstall with labels(os=linux,arch=amd64): SKIP
All Checks passed

$ krew audit foo
....
```

## Usage for Developers and CI

The plugin developer can invoke the command locally. In the CI we can call the command for all updated plugin files like:

```bash
for name in $(git diff --name-only HEAD~1 HEAD | grep -o "^plugins/.*.yaml"); do
  krew audit "$name"
done
```

The CI executes this command for different platforms to test the installation.

The audit command should check for:
* Manifest
* Naming conventions
* Installation & Removal

fixes https://github.com/GoogleContainerTools/krew-index/issues/15

**Comment by ahmetb** [comment]
I like the idea. I think it should not be a user-visible command that pollutes "kubectl plugin --help" surface. For example, `brew audit` is hidden by default. But I think we can't do that just yet.

How about making this a separate plugin that only plugin devs need to install? This way the audit can evolve/live independently.

On the CI side, I think we need a command that can download and install the plugin for ALL os/arch combinations. So something in the lines of:

    <cmd> audit --source=<plugin>.yaml [--os=] [--arch=] [--target-dir=$KREW_INSTALL_PATH?]

**Comment by ahmetb** [comment]
So we solved the --os/--arch customization with KREW_OS and KREW_ARCH in #70. Similarly, --target-dir= was solved in #65.

The primary challenge I'm facing right now is figuring out all the supported os/arch matrices from `platform.selector` to write a cmd that executes

    KREW_OS=... KREW_ARCH=... KREW_ROOT=$(mktemp -d) krew install --source=<plugin.yaml>

to verify all platform urls/fileOps are working.

**Comment by ahmetb** [comment]
Postponing to after v0.2.

**Comment by ahmetb** [comment]
Fixed with #151 by adding a manifest validator tool.
